x-pack/plugin/snapshot-based-recoveries/src/internalClusterTest/java/org/elasticsearch/xpack/snapshotbasedrecoveries/recovery/SnapshotBasedIndexRecoveryIT.java

```diff
@@ -736,6 +737,15 @@ public void sendResponse(Exception exception) throws IOException {
                                     });
                                 }
                             }
+
+                            private boolean assertShardClosedException(Exception e) {
+                                if (e instanceof CancellableThreads.ExecutionCancelledException cancelledException) {
+                                    assert cancelledException.getMessage().contains("shard closed") : e;
+                                } else {
+                                    assert e instanceof IndexShardClosedException : e;
+                                }
```

Comment:
NIT: could be simplified with:
```
                                assert (e instanceof CancellableThreads.ExecutionCancelledException cancelledException && cancelledException.getMessage().contains("shard closed"))
                                    || e instanceof IndexShardClosedException : e;
```
or many be there is even better `assertThat` candidate since this is IT?

Reply:
I tried to keep the change as minimal as possible (and I also find the 2 lines more readable).

I hope it's OK if I merge in current state.